website/docs/tutorials/07-testing/02-cheatcodes/expect-call.md

```diff
@@ -2,7 +2,9 @@
 ```python
 def expect_call(self, contract_address: int, fn_name: str, calldata: list[int]) -> None: ...
 ```
-Allows to specify what calls to contracts' functions are expected in the code. You can use this cheatcode in any place in the code as it checks for the desired calls' existence at the very end of the program's execution.
+Allows to specify what calls to contracts' functions are expected in the code.
+It checks for the expected call's existence until the returned callable is called.
+If the returned callable is not invoked, the expected call is being awaited until the end of the scope.
```

Comment:
This cheatcode doesn't check for call's existence. It sets the call expectation. The returned callback checks whether the call was made or not.

Reply:
[update docs](https://github.com/software-mansion/protostar/pull/1213/commits/21bec7875f678068bf55e1b03705fe0cb2995d05)